### PR TITLE
more clean up

### PR DESCRIPTION
**Pull request by zhexuany**
adding `SupressWarning("unchecked")` which can suppress `unchecked` warning in our code base.

---
This change is [Reviewable](https://reviewable.io/reviews/pingcap/tikv-client-lib-java/84)

### CODE REVIEW

**Comment by zhexuany** [comment]
Another thing I did in this PR is migrating `Comparables` to `TiKey`.

File: src/main/java/com/pingcap/tikv/meta/TiKey.java [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+  @SuppressWarnings("unchecked")
+  public int compareTo(@Nonnull TiKey<T> o) {
+    if (data instanceof Comparable) {
+      return ((Comparable<T>)data).compareTo(o.data);
+    } else if (data instanceof byte[]) {
+      return compareTo((byte[]) o.data);
+    } else if (data instanceof ByteString) {
+      return compareTo((ByteString) o.data);
+    }
+    return 0;
```
Comment (birdstorm) [inline src/main/java/com/pingcap/tikv/meta/TiKey.java:70]: Why does compareTo return 0 here?
I think we should throw some kind of exception instead?

File: src/main/java/com/pingcap/tikv/meta/TiKey.java [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+  private static final Comparator<byte[]> comparator = UnsignedBytes.lexicographicalComparator();
+  private T data;
+
+  public TiKey(@Nonnull T data) {
+    this.data = data;
+  }
+
+  private int compareTo(@Nonnull ByteString o) {
+    requireNonNull(o, "other is null");
+    ByteString data = (ByteString)this.data;
```
Comment (birdstorm) [inline src/main/java/com/pingcap/tikv/meta/TiKey.java:41]: maybe we should add <instanceof> if-clause to deal with exceptions?
Here we have dealt with TiKey's comparison against Comparable, ByteString and byte[], but not yet resolved cases vice versa. 
The same goes following two compareTo functions.

**Comment by birdstorm** [comment]
One of the by-effects of implementing TiKey is that we would have to `new TiKey<>()` for a new `TiKey` object wherever comparison with constant — or any type we want to wrap — is needed, which doesn't really improve the preceding situation where `Comparables.wrap()` are used too much. 
It also seems that it would be difficult for us to change its constructor into a static one, eg., `public static TiKey<T> create()` while using `TiKey.create()`, which would be safer and more elegant than `new TiKey<>()`.
On the contrary, the good part is that we will not be dealing with annoying unchecked warnings. For now, I think we should first make the code readable and robust for all input data types.

**Comment by zhexuany** [comment]
too much conflicts. Closing.

**Pull request closed without merging**